src/wrappers/themis/Obj-C/objcthemis/ssession_transport_interface.h

```diff
@@ -19,15 +19,13 @@
 * @brief secure session trancport callbacs interface
 */
 #import <Foundation/Foundation.h>
-#import <themis/themis.h>
+
+typedef struct secure_session_user_callbacks_type secure_session_user_callbacks_t;
```

Comment:
i dunno what is the better way to remove header, but to expose callback, and not to break backwards-compatibility for current users.

Reply:
Do we need to avoid _all and any_ `<themis/>` headers? If not then I'd think that a proper way to avoid duplication would be to introduce some (private) `<themis/types.h>` header to keep all our typedefs and include it here.

Though since we have only this typedef to take care of I think it is fine to leave it as is. If we decide to remove or rename the underlying structure then ssession_transport_interface.m will stop compiling so this typedef can't get out of sync with the main header. (And I guess the compiler will warn us about conflicting typedefs if they do not match.)

However, I think that Obj-C users have never actually used the callback struct (as they don't have any reason to). So I guess this typedef can be safely removed. Since `_callbacks` ivar is no longer a public, any code that actually used it should fail to compile now — this _is_ a breaking change, I'd imagine.

Reply:
`_callbacks` is not a public property, and it's not supposed to be accessed "as is", so I safely moved it into `.m` file here

https://github.com/cossacklabs/themis/pull/415/files/21bc76aa0a906b44e5542d9ed4a5acf6a18fcd6b#diff-a5dd6093239a6c2087d20d0b49978d9fR22

Basically, `_callbacks` are _set_ during [init](https://github.com/cossacklabs/themis/pull/415/files/21bc76aa0a906b44e5542d9ed4a5acf6a18fcd6b#diff-a5dd6093239a6c2087d20d0b49978d9fL51) only.

`_callbacks` _getter_ is a method `- (secure_session_user_callbacks_t *)callbacks;`

And you're right, these things are used in SecureSession wrapper only ([here](https://github.com/cossacklabs/themis/pull/415/files/21bc76aa0a906b44e5542d9ed4a5acf6a18fcd6b#diff-8fc02a611466e238c5a7c46f2a2b41b5R35)), users don't use them, so I believe it's not a breaking changes.